rudder-agent/SOURCES/check-rudder-agent

```diff
@@ -241,3 +241,12 @@ if [ "z${OS_FAMILY}" != "zAIX" ]; then
   check_and_fix_specific_rudder_agent_file /etc/default/rudder-agent default
   check_and_fix_specific_rudder_agent_file /etc/cron.d/rudder-agent cron
 fi
+
+# if file is absent or empty there have been a problem with update
+base="/var/rudder/cfengine-community/inputs"
+if [ ! -f "${base}/common/1.0/update.cf" ] || [ $(stat -c "%s" "${base}/common/1.0/update.cf") = 0 ] || [ ! -f "${base}/failsafe.cf" ] || [ $(stat -c "%s" "${base}/failsafe.cf") = 0 ]
```

Comment:
you ought to check promises.cf; if it is not there, cf-execd may never run and restore the situation